oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/AnalyzerSetting.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.elasticsearch.base;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class AnalyzerSetting {
+    /**
+     * A built-in or customised tokenizer.
+     */
+    private Map<String, Object> tokenizer = new HashMap<>();
+    /**
+     * An optional array of built-in or customised character filters.
+     */
+    @SerializedName("char_filter")
+    private Map<String, Object> charFilter = new HashMap<>();
+    /**
+     * An optional array of built-in or customised token filters.
+     */
+    private Map<String, Object> filter = new HashMap<>();
+    /**
+     * The custom analyzers.
+     */
+    private Map<String, Object> analyzer = new HashMap<>();
+
+    public void combine(AnalyzerSetting analyzerSetting) {
```

Comment:
Why do we need `combine`? Could you share the use case?

Reply:
![image](https://user-images.githubusercontent.com/31562192/105634788-c7809780-5e9a-11eb-9204-bc887ad53267.png)
one index setting has multi analyzers